core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -1224,12 +1221,25 @@ private SchemaKey getSchemaKeyUsingContexts(int id, String subject)
         QualifiedSubject qualSub =
             new QualifiedSubject(v.getTenant(), v.getContext(), qs.getSubject());
         SchemaKey key = lookupCache.schemaKeyById(id, qualSub.toQualifiedSubject());
-        if (key != null) {
+        if (schemaKeyMatchesSubject(key, qualSub)) {
           return key;
         }
       }
     }
-    return null;
+    // Could not find the id in subjects in other contexts,
+    // just return the id in the default context if found
+    return subjectVersionKey;
+  }
+
+  private boolean schemaKeyMatchesSubject(SchemaKey key, QualifiedSubject qs) {
+    if (key == null) {
+      return false;
+    } else if (qs == null) {
+      return true;
```

Comment:
Is it correct that `qs` can only be null if `subject` is null so in theory this branch will never be entered because of the short circuit of `subject == null`?

Reply:
You are correct.  I just added the check as a defensive check in case this method gets used elsewhere in the future.